### PR TITLE
Task/devdocs ci refresh

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR updates the RAJA Dev Guide sections that describe our CI, mostly the GitLab CI part.

The generated documentation can be viewed here: https://raja.readthedocs.io/en/task-devdocs-ci-refresh/sphinx/dev_guide/index.html

A few of you asked about a walk through of our GitLab stuff. Should we schedule a meeting for that?

### CODE REVIEW

File: docs/sphinx/dev_guide/build_configurations.rst [hunk 1]
```diff
@@ -152,7 +152,14 @@ Generating a host-config file
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 To generate a host-config file for a desired configuration, run the 
-``uberenv.py`` python script from the top-level RAJA directory. For example,
+``uberenv.py`` python script from the top-level RAJA directory. 
+
+.. important:: **Do not** run the ``uberenv.py`` script, which invokes Spack
+               in your LC home directory since Spack uses a lot of disk space.
+               Running Spack in your home directory may cause you to exceed
+               your LC disk space quota.
```
Comment (MrBurmark) [inline docs/sphinx/dev_guide/build_configurations.rst:160]:
```suggestion
.. important:: **Do not** run the ``uberenv.py`` script, which invokes Spack,
               in your LC home directory.
               Running Spack in your home directory may cause you to exceed
               your LC disk space quota.
```

File: docs/sphinx/dev_guide/ci.rst [hunk 2]
```diff
@@ -151,25 +161,27 @@ support LC GitLab CI testing.
...
-    files that are included by this one. Global variables can also be defined 
-    here.
-  * The ``.uberenv_config.json`` file defines the Spack version we use, where 
-    Spack packages live, etc.
-  * Files in the ``RAJA/.gitlab`` directory define test pipelines that RAJA
-    subscribes to an which are defined in the 
+Briefly, these files play the following roles in GitLab CI testing:
+
+  * The `RAJA/.gitlab-ci.yml <https://github.com/LLNL/RAJA/tree/develop/.gitlab-ci.yml>`_ file is the top-level file for GitLab CI configuration. It defines 
+    variables used in all GItLab pipelines such as
```
Comment (MrBurmark) [inline docs/sphinx/dev_guide/ci.rst:170]:
```suggestion
    variables used in all GitLab pipelines such as
```

File: docs/sphinx/dev_guide/ci.rst [hunk 3]
```diff
@@ -151,25 +161,27 @@ support LC GitLab CI testing.
...
+    GitHub project name and organization, service user account name, version
+    information for RADIUSS Shared CI project we are using, and 
+    top-level information for triggering build-and-test sub-pipelines.
+  * The `RAJA/.uberenv_config.json <https://github.com/LLNL/RAJA/tree/develop/.uberenv_config.json>`_ file defines information about Spack such as
+    Spack version we are using, location of Spack packages, etc.
+  * The `RAJA/.gitlab <https://github.com/LLNL/RAJA/tree/develop/.gitlab>`_ 
+    directory contains several files that connect RAJA GitLab pipelines to 
+     shared pipelines defined in the
```
Comment (MrBurmark) [inline docs/sphinx/dev_guide/ci.rst:178]:
```suggestion
    shared pipelines defined in the 
```

File: docs/sphinx/dev_guide/ci.rst [hunk 4]
```diff
@@ -261,121 +258,47 @@ Running a CI build and test pipeline  (steps 3, 4, 5, 6)
...
-
-Lastly, test results are collected in a JUnit XML file that
-GitLab uses for reporting the results in its GUI **(step 6)**. This is
-done by the 
-`RADIUSS Shared CI Framework <https://github.com/LLNL/radiuss-shared-ci>`_
-
-The commands shown here intermingle with other commands that emit messages,
-timing information for various operations, etc. which appear in a log
-file that can be viewed in the GitLab GUI.
+Also, recall that to generate a hist-config file, Spack uses packages and
```
Comment (MrBurmark) [inline docs/sphinx/dev_guide/ci.rst:283]:
```suggestion
Also, recall that to generate a host-config file, Spack uses packages and 
```

File: docs/sphinx/dev_guide/build_configurations.rst (hunk 1 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/build_configurations.rst]: Fixed.

File: docs/sphinx/dev_guide/ci.rst (hunk 2 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst]: Fixed.

File: docs/sphinx/dev_guide/ci.rst (hunk 3 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst]: Fixed.

File: docs/sphinx/dev_guide/ci.rst (hunk 4 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst]: Fixed.

File: docs/sphinx/dev_guide/ci.rst [hunk 5]
```diff
@@ -17,28 +17,24 @@ Continuous Integration (CI) Testing
...
-    builds for CUDA, HIP, and SYCL on Azure, RAJA tests are only run for 
-    CPU-only pipelines. Docker container images we use for the Linux testing
-    are maintained in the 
-    `RSE Ops Project <https://github.com/rse-ops/docker-images>`_. Please see 
-    the `RAJA Azure DevOps <https://dev.azure.com/llnl/RAJA>`_ project to learn 
-    more about our testing there.
-
-  * **GitLab** instances in the Collaboration Zone (CZ) of the Livermore 
-    Computing (LC) Center run builds and tests on LC HPC platforms using
+    environments using compilers in container image maintained in the
```
Comment (rchen20) [inline docs/sphinx/dev_guide/ci.rst:23]:
```suggestion
    environments using compilers in container images maintained in the
```

File: docs/sphinx/dev_guide/ci.rst [hunk 6]
```diff
@@ -17,28 +17,24 @@ Continuous Integration (CI) Testing
...
+    `RSE Ops Project <https://github.com/rse-ops/docker-images>`_.
+    While we do some GPU builds on Azure, RAJA tests are only run for CPU-only 
+    builds. The current set of builds run on Azure can be seen by looking at
+    the ``RAJA/azure-pipelines.yml`` and ``RAJA/Dockerfile`` files.
+
+  * **GitLab** instance in the Collaboration Zone (CZ) of the Livermore
```
Comment (rchen20) [inline docs/sphinx/dev_guide/ci.rst:29]:
```suggestion
  * **GitLab** instances in the Collaboration Zone (CZ) of the Livermore 
```

File: docs/sphinx/dev_guide/ci.rst (hunk 5 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:23]: Fixed.

File: docs/sphinx/dev_guide/ci.rst (hunk 6 above)
Comment (rhornung67) [reply docs/sphinx/dev_guide/ci.rst:29]: There is only one GitLab instance.

Comment (rchen20) [reply docs/sphinx/dev_guide/ci.rst:29]: I was confused by `run`, but I see you changed it to `runs`.

**Comment by rhornung67** [comment]
Can I get some reviews on this? Thanks.

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**